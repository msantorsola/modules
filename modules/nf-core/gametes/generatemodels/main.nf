process GAMETES_GENERATEMODELS {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gametes:2.1--py310h7cba7a3_0':
        'biocontainers/gametes:2.1--py310h7cba7a3_0' }"

    input:
    tuple val(meta), val(heritability)

    output:
    tuple val(meta), path("_EDM_Scores.txt"), emit: edm_scores, optional: true
    tuple val(meta), path("*_OddsRatio_Scores.txt"), emit: oddsratio_scores, optional: true
    tuple val(meta), path("*_Models.txt"), emit: models
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '2.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.

    """
    gametes \\
        -M  "\\
        -o $prefix \\
        $args"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gametes: $VERSION
    END_VERSIONS
    """
    
    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '2.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    
    """
    touch ${prefix}_EDM_Scores.txt
    touch ${prefix}_OddsRatio_Scores.txt
    touch ${prefix}_Models.txt


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
       gametes: $VERSION
    END_VERSIONS
    """

}

