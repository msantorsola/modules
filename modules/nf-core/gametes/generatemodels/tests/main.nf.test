nextflow_process {

    name "Test Process GAMETES_GENERATEMODELS"
    script "../main.nf"
    process "GAMETES_GENERATEMODELS"

    tag "modules"
    tag "modules_nfcore"
    tag "gametes"
    tag "gametes/generatemodels"
    //config "./nextflow.config"

    test("dataset - model") {


        when {
            process {
                """

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("https://raw.githubusercontent.com/msantorsola/random_tests/main/gametesM_dict.yml"),
                ]
                """
            }
        }


        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}