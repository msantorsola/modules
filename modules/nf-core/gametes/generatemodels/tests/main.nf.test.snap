{
    "dataset - model": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "edm_scores": [
                    
                ],
                "models": [
                    
                ],
                "oddsratio_scores": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-20T17:18:46.297903766"
    }
}